ivy/functional/frontends/numpy/indexing_routines/indexing_like_operations.py

```diff
@@ -0,0 +1,10 @@
+import ivy
+
+
+def diagonal(x, offset, axis1, axis2):
+    ivy.assertions.check_equal(
+        axis1 == axis2, message="Both axis values should not be the same"
+    )
+    ivy.assertions.check_equal(x.ndim, 2, message="x must be 2-dimensional")
+
```

Comment:
You should remove the assertions from here. Assertions are triggered already by the backend functions. 

Also, you should make sure that the test does not trigger assertions, i.e. the test should only provide valid arguments to the function.

Reply:
ok.